Build broken on gcc 8.2.0

[648/815] Building C object libshaderc_spvc/CMakeFiles/spvc_c_smoke_test.dir/src/spvc_smoke_test_util.c.o
cc1: warning: command line option ‘-Wextra-semi’ is valid for C++/ObjC++ but not for C
[649/815] Building C object libshaderc_spvc/CMakeFiles/spvc_c_smoke_test.dir/src/spvc_c_smoke_test.c.o
cc1: warning: command line option ‘-Wextra-semi’ is valid for C++/ObjC++ but not for C
[650/815] Building C object libshaderc_spvc/CMakeFiles/spvc_webgpu_c_smoke_test.dir/src/spvc_smoke_test_util.c.o
cc1: warning: command line option ‘-Wextra-semi’ is valid for C++/ObjC++ but not for C
[652/815] Building C object libshaderc_spvc/CMakeFiles/spvc_webgpu_c_smoke_test.dir/src/spvc_webgpu_c_smoke_test.c.o
cc1: warning: command line option ‘-Wextra-semi’ is valid for C++/ObjC++ but not for C
[680/815] Building C object libshaderc/CMakeFiles/shaderc_c_smoke_test.dir/src/shaderc_c_smoke_test.c.o
cc1: warning: command line option ‘-Wextra-semi’ is valid for C++/ObjC++ but not for C
[700/815] Building CXX object libshaderc_spvc/CMakeFiles/shaderc_spvc.dir/src/spvc.cc.o
FAILED: libshaderc_spvc/CMakeFiles/shaderc_spvc.dir/src/spvc.cc.o 
/usr/bin/c++  -DNV_EXTENSIONS -I../../libshaderc_spvc/include -I../../libshaderc/include -I../../libshaderc_util/include -I../../third_party/spirv-tools/include -I../../third_party/spirv-cross/.. -I../../third_party/spirv-headers/include -I../../third_party/spirv-cross -Wimplicit-fallthrough -g   -Wextra-semi -Wall -Werror -fvisibility=hidden -fPIC -std=c++11 -std=gnu++11 -MD -MT libshaderc_spvc/CMakeFiles/shaderc_spvc.dir/src/spvc.cc.o -MF libshaderc_spvc/CMakeFiles/shaderc_spvc.dir/src/spvc.cc.o.d -o libshaderc_spvc/CMakeFiles/shaderc_spvc.dir/src/spvc.cc.o -c ../../libshaderc_spvc/src/spvc.cc
../../libshaderc_spvc/src/spvc.cc: In function ‘size_t shaderc_spvc_compile_options_set_for_fuzzing(shaderc_spvc_compile_options_t, const uint8_t*, size_t)’:
../../libshaderc_spvc/src/spvc.cc:245:41: error: ‘void* memcpy(void*, const void*, size_t)’ writing to an object of type ‘struct shaderc_spvc_compile_options’ with no trivial copy-assignment; use copy-assignment or copy-initialization instead [-Werror=class-memaccess]
   memcpy(options, data, sizeof(*options));
                                         ^
../../libshaderc_spvc/src/spvc.cc:37:8: note: ‘struct shaderc_spvc_compile_options’ declared here
 struct shaderc_spvc_compile_options {
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
[712/815] Building CXX object libshaderc_spvc/CMakeFiles/shaderc_spvc_shared.dir/src/spvc.cc.o
FAILED: libshaderc_spvc/CMakeFiles/shaderc_spvc_shared.dir/src/spvc.cc.o 
/usr/bin/c++  -DNV_EXTENSIONS -DSHADERC_IMPLEMENTATION -DSHADERC_SHAREDLIB -Dshaderc_spvc_shared_EXPORTS -I../../libshaderc_spvc/include -I../../libshaderc/include -I../../libshaderc_util/include -I../../third_party/spirv-tools/include -I../../third_party/spirv-cross/.. -I../../third_party/spirv-headers/include -I../../third_party/spirv-cross -Wimplicit-fallthrough -g -fPIC   -Wextra-semi -Wall -Werror -fvisibility=hidden -fPIC -std=c++11 -std=gnu++11 -MD -MT libshaderc_spvc/CMakeFiles/shaderc_spvc_shared.dir/src/spvc.cc.o -MF libshaderc_spvc/CMakeFiles/shaderc_spvc_shared.dir/src/spvc.cc.o.d -o libshaderc_spvc/CMakeFiles/shaderc_spvc_shared.dir/src/spvc.cc.o -c ../../libshaderc_spvc/src/spvc.cc
../../libshaderc_spvc/src/spvc.cc: In function ‘size_t shaderc_spvc_compile_options_set_for_fuzzing(shaderc_spvc_compile_options_t, const uint8_t*, size_t)’:
../../libshaderc_spvc/src/spvc.cc:245:41: error: ‘void* memcpy(void*, const void*, size_t)’ writing to an object of type ‘struct shaderc_spvc_compile_options’ with no trivial copy-assignment; use copy-assignment or copy-initialization instead [-Werror=class-memaccess]
   memcpy(options, data, sizeof(*options));
                                         ^
../../libshaderc_spvc/src/spvc.cc:37:8: note: ‘struct shaderc_spvc_compile_options’ declared here
 struct shaderc_spvc_compile_options {
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors

--- a/libshaderc_spvc/src/spvc.cc
+++ b/libshaderc_spvc/src/spvc.cc
@@ -242,7 +242,7 @@ size_t shaderc_spvc_compile_options_set_for_fuzzing(
     shaderc_spvc_compile_options_t options, const uint8_t* data, size_t size) {
   if (!data || size < sizeof(*options)) return 0;
 
-  memcpy(options, data, sizeof(*options));
+  memcpy(static_cast<void*>(options), data, sizeof(*options));
   return sizeof(*options);
 }
 

